sklearnex/neighbors/knn_classification.py

```diff
@@ -15,4 +15,156 @@
+        if X is not None:
+            X = _check_array(
+                X, accept_sparse='csr', dtype=[
+                    np.float64, np.float32])
+
+        if onedal_model is not None and not sp.issparse(X):
+            result = onedal_KNeighborsMixin._kneighbors(self, X, n_neighbors, return_distance)
+
+        return result
+
+if LooseVersion(sklearn_version) >= LooseVersion("0.24"):
+    class KNeighborsClassifier_(sklearn_NeighborsBase, KNeighborsMixin, onedal_KNeighborsClassifier):
+        @_deprecate_positional_args
+        def __init__(self, n_neighbors=5, *,
+                     weights='uniform', algorithm='auto', leaf_size=30,
+                     p=2, metric='minkowski', metric_params=None, n_jobs=None,
+                     **kwargs):
+            super().__init__(
+                n_neighbors=n_neighbors,
+                algorithm=algorithm,
+                leaf_size=leaf_size, metric=metric, p=p,
+                metric_params=metric_params,
+                n_jobs=n_jobs, **kwargs)
+            self.weights = \
+                weights if LooseVersion(sklearn_version) >= LooseVersion("1.0") else _check_weights(weights)
+elif LooseVersion(sklearn_version) >= LooseVersion("0.22"):
+    from sklearn.neighbors._base import SupervisedIntegerMixin as \
+        BaseSupervisedIntegerMixin
+
+    class KNeighborsClassifier_(sklearn_NeighborsBase, onedal_KNeighborsClassifier, KNeighborsMixin,
+                                BaseSupervisedIntegerMixin):
+        @_deprecate_positional_args
+        def __init__(self, n_neighbors=5, *,
+                     weights='uniform', algorithm='auto', leaf_size=30,
+                     p=2, metric='minkowski', metric_params=None, n_jobs=None,
+                     **kwargs):
+            super().__init__(
+                n_neighbors=n_neighbors,
+                algorithm=algorithm,
+                leaf_size=leaf_size, metric=metric, p=p,
+                metric_params=metric_params,
+                n_jobs=n_jobs, **kwargs)
+            self.weights = _check_weights(weights)
+else:
+    from sklearn.neighbors.base import SupervisedIntegerMixin as \
+        BaseSupervisedIntegerMixin
+
+    class KNeighborsClassifier_(sklearn_NeighborsBase, onedal_KNeighborsClassifier, KNeighborsMixin,
+                                BaseSupervisedIntegerMixin):
+        @_deprecate_positional_args
+        def __init__(self, n_neighbors=5, *,
+                     weights='uniform', algorithm='auto', leaf_size=30,
+                     p=2, metric='minkowski', metric_params=None, n_jobs=None,
+                     **kwargs):
+            super().__init__(
+                n_neighbors=n_neighbors,
+                algorithm=algorithm,
+                leaf_size=leaf_size, metric=metric, p=p,
+                metric_params=metric_params,
+                n_jobs=n_jobs, **kwargs)
+            self.weights = _check_weights(weights)
+
+class KNeighborsClassifier(KNeighborsClassifier_):

```

Comment:
It shouldn't be inherited from onedal classes because their semantic is different from scikit. Use composition instead